packages/astro/src/core/render/dev/index.ts

```diff
@@ -103,7 +104,7 @@ export async function render(renderers: Renderer[], mod: ComponentInstance, ssrO
 			// broken in the legacy build. This can be removed once the legacy build is removed.
 			if (!astroConfig.buildOptions.legacyBuild) {
 				const [, resolvedPath] = await viteServer.moduleGraph.resolveUrl(s);
-				return resolvedPath;
+				return '/@fs' + prependForwardSlash(resolvedPath);
```

Comment:
Vite's module graph gives us a broken URL? That's so strange.

Reply:
The problem is that there isn't a function in Vite to give a module ID and return a client-compatible pathname. resolveUrl is the closest there is.

Reply:
fyi @matthewp and I chatted about this a bit earlier and my integration work might improve this a bit more. The first return value of `resolveUrl()` is sometimes a more correct URL that we might be able to leverage for windows